File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/views/HttpCallSearchView.java
package com.prateekj.snooper.networksnooper.views;

import com.prateekj.snooper.networksnooper.model.HttpCallRecord;

import java.util.List;

public interface HttpCallSearchView {
  void showResults(List<HttpCallRecord> httpCallRecords);
  void showNoResultsFoundMessage(String keyword);
  void hideSearchResultsView();
  void showLoader();
  void hideLoader();
}

File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/views/HttpCallSearchView.kt
package com.prateekj.snooper.networksnooper.views

import com.prateekj.snooper.networksnooper.model.HttpCallRecord

interface HttpCallSearchView {
  fun showResults(httpCallRecords: List<HttpCallRecord>)
  fun showNoResultsFoundMessage(keyword: String)
  fun hideSearchResultsView()
  fun showLoader()
  fun hideLoader()
}
